parts/windows/windowscalicofunc.ps1

```diff
@@ -27,10 +27,27 @@ function GetCalicoKubeConfig {
         [parameter(Mandatory=$false)] $KubeConfigPath = "c:\\k\\config"
     )
 
-    $name=c:\k\kubectl.exe --kubeconfig=$KubeConfigPath get secret -n $CalicoNamespace --field-selector=type=kubernetes.io/service-account-token --no-headers -o custom-columns=":metadata.name" | findstr $SecretName | select -first 1
+    # When creating Windows agent pools with the system Linux agent pool, the service account for calico may not be available in provisioning Windows agent nodes.
+    # So we need to wait here until the service account for calico is available
+    $name=""
+    $retryCount=0
+    $retryInterval=5
+    $maxRetryCount=120 # 10 minutes
+
+    do {
+        $name=c:\k\kubectl.exe --kubeconfig=$KubeConfigPath get secret -n $CalicoNamespace --field-selector=type=kubernetes.io/service-account-token --no-headers -o custom-columns=":metadata.name" | findstr $SecretName | select -first 1

```

Comment:
If it's possible to return the message error? to be more debuggable? Because an empty stdout may be caused by other issues like network failure or apiserver unavailability.
And how can we recover from this error if 10 minutes is not enough?

Reply:
+1 for logging errors

Reply:
I totally agree with all of you. We have a task for this but this needs more time to do it. For now, it is not supported. We only can collect `CustomDataSetupScript.log` for troubleshooting.

Reply:
@mainred We cannot recover it when provisioning fails for this error or other errors. There is no difference here. We can consider to improve it as a feature later.

Reply:
It does not return any error here. it only cannot find the specified $SecretName